app/src/main/java/org/oppia/android/app/activity/ActivityRouter.kt

```diff
@@ -0,0 +1,31 @@
+package org.oppia.android.app.activity
+
+import androidx.appcompat.app.AppCompatActivity
+import org.oppia.android.app.home.recentlyplayed.RecentlyPlayedActivity
+import org.oppia.android.app.model.DestinationScreen
+import org.oppia.android.app.model.RecentlyPlayedActivityIntentExtras
+import javax.inject.Inject
+
+/** Central utility to manage routing to different activities. */
+class ActivityRouter @Inject constructor(private val activity: AppCompatActivity) {
+
+  /** Checks the value of [DestinationScreen] and routes to different activities accordingly. */
+  fun routeToScreen(destinationScreen: DestinationScreen) {
+    if (destinationScreen.destinationScreenCase ==
```

Comment:
Should this be a ``when`` statement,  instead? Maybe also use 'return' to force it to be exhaustive.